crates/nu-parser/src/parser.rs

```diff
@@ -4919,7 +4926,14 @@ pub fn parse_closure_expression(
         }
     }
 
-    let mut output = parse_block(working_set, &output[amt_to_skip..], span, false, false);
+    let mut output = parse_block(
+        working_set,
+        &output[amt_to_skip..],
+        span,
+        false,
+        false,
+        true,
```

Comment:
With all those bools, is this the point where a self-documenting `enum BlockKind` makes sense? The closure and regular block seem to have mutually exclusive patterns and only the `is_scoped` thing appears for blocks as a variation.

Reply:
that makes sense. to be honest i was hoping someone with more parser experience would have some magic solution that could get around having to pass whether the block is a closure or not, but this is reasonable as well.